x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/connector/attach_index_api_logic.ts

```diff
@@ -11,15 +11,21 @@ import { HttpLogic } from '../../../shared/http';
 export interface AttachIndexApiLogicArgs {
   connectorId: string;
   indexName: string;
+  isNative: boolean;
```

Comment:
You're now adding this isNative flag in a bunch of different files in the frontend. If we instead just determine whether a connector is native in the Kibana backend route, we only need to check it there--and that'd lead to more consistent results (we wouldn't have situations where people convert a native connector then generate a secret for it anyway because of cache misalignment).

Reply:
I replaced this with calling `fetchConnectorById` and using that result to determine if it's native